chart/flux/values.yaml

```diff
@@ -298,3 +298,10 @@ hostAliases: {}
 #    hostnames:
 #    - "foo.remote"
 #    - "bar.remote"
+
+podSecurityContext: {}
+#  allowPrivilegeEscalation: false
+#  fsGroup: 1001
+#  runAsUser: 1001
+#  runAsGroup: 1001
+#  runAsNonRoot: true
```

Comment:
None of these work with Flux as it runs as root.

Reply:
It forces the container to run as non-root. I couldn't find in the documentation any reason for flux to run as root, so I try to run it as a normal user.
```
docker run --rm --user 1001 docker.io/fluxcd/flux:1.19.0 
ts=2020-05-11T09:38:53.944207902Z caller=main.go:259 version=1.19.0
```
It could work, maybe it needs additional emptyDir to work with git pulled directories, to avoid permission issues

Reply:
Making Flux run as non-root is a non-trivial task. The SSH keygen, secrets mounts and gpg setup need to be rethink and addressed both in code and configs.

See:
- https://github.com/fluxcd/flux/issues/1327
- https://github.com/fluxcd/helm-operator/issues/233

Reply:
Thanks for this explanation and links.
I run Flux only with git over HTTPS so I do not face any of theses issues. I understand it needs to stay backwards compatible and therefor can't be merged as-is.
I will check if I can help on pushing some other code to get it working.

Reply:
Hi @alemairebe, can pls explain how you do that "Flux only with git over HTTPS so I do not face any of theses issues". Thx

Reply:
Hi @marmelin ,
I do not use SSH keys nor GPG, I pass git https credentials from a secret with the helm chart value : "env.secretName" then on "git.url" I have "https://$(username):$(password)@my_git_repo.local/repo_project/my_repo.git" where $(username) and $(password) are fetched form the secret.
This is well described in the documentation : https://github.com/fluxcd/flux/tree/master/chart/flux#flux-with-git-over-https

Reply:
Do you run flux with root or non-root? Because I thought root is only required for ssh git connection.

Reply:
I run flux with `runAsUser: 1001` could be any number but 0